testing/monkey/monkey.c

```diff
@@ -0,0 +1,196 @@
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <stdlib.h>
+#include <string.h>
+#include <inttypes.h>
+#include "monkey.h"
+#include "monkey_recorder.h"
+#include "monkey_assert.h"
+#include "monkey_log.h"
+#include "monkey_dev.h"
+#include "monkey_utils.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define MONKEY_DEV_CREATE_MATCH(monkey, type_mask, type)                  \
+do {                                                                      \
+  if ((type_mask & MONKEY_DEV_TYPE_##type) == MONKEY_DEV_TYPE_##type)     \
+    {                                                                     \
+      FAR struct monkey_dev_s *dev;                                       \
+      dev = monkey_dev_create(CONFIG_TESTING_MONKEY_DEV_PATH_##type,      \
+                              MONKEY_DEV_TYPE_##type);                    \
+      if (!dev)                                                           \
+        {                                                                 \
+          goto failed;                                                    \
+        }                                                                 \
+      monkey->devs[monkey->dev_num] = dev;                                \
+      monkey->dev_num++;                                                  \
+    }                                                                     \
+} while (0)
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: monkey_create
+ ****************************************************************************/
+
+FAR struct monkey_s *monkey_create(int dev_type_mask)
+{
+  FAR struct monkey_s *monkey = malloc(sizeof(struct monkey_s));
+  MONKEY_ASSERT_NULL(monkey);
+  memset(monkey, 0, sizeof(struct monkey_s));
```

Comment:
Optional: `zalloc` or `calloc` can used to remove `memset`

Reply:
done